roles/translator/src/main.rs

```diff
@@ -77,6 +81,129 @@ async fn main() {
+            interrupt_signal = interrupt_signal_future => {
+                match interrupt_signal {
+                    Ok(()) => {
+                        info!("Interrupt received");
+                    },
+                    Err(err) => {
+                        error!("Unable to listen for interrupt signal: {}", err);
+                        // we also shut down in case of error
+                    },
+                }
+                break;
+            }
+        };
+        let task_status: Status = task_status.unwrap();
+
+        match task_status.state {
+            // Should only be sent by the downstream listener
+            State::DownstreamShutdown(err) => {
+                error!("SHUTDOWN from: {}", err);
+                break;
+            }
+            State::BridgeShutdown(err) => {
+                error!("SHUTDOWN from: {}", err);
+                break;
+            }
+            State::UpstreamShutdown(err) => {
+                error!("SHUTDOWN from: {}", err);
+
+                // wait a random amount of time between 0 and 3000ms
+                // if all the downstreams try to reconnect at the same time, the upstream may fail
+                tokio::time::sleep(Duration::from_millis(1000)).await;
+                let mut rng = rand::thread_rng();
+                let wait_time = rng.gen_range(0..=3000);
+                tokio::time::sleep(Duration::from_millis(wait_time)).await;
+
+                // kill al the tasks
+                let task_collector_aborting = task_collector_.clone();
+                kill_tasks(task_collector_aborting.clone());
+
+                error!("Trying recconnecting to upstream");
+                start(
+                    rx_sv2_submit_shares_ext.clone(),
+                    tx_sv2_submit_shares_ext.clone(),
+                    tx_sv2_new_ext_mining_job.clone(),
+                    tx_sv2_set_new_prev_hash.clone(),
+                    tx_sv2_extranonce.clone(),
+                    rx_sv2_extranonce.clone(),
+                    rx_sv2_set_new_prev_hash.clone(),
+                    rx_sv2_new_ext_mining_job.clone(),
+                    rx_sv1_downstream.clone(),
+                    tx_sv1_bridge.clone(),
+                    tx_sv1_notify.clone(),
+                    target.clone(),
+                    tx_status.clone(),
+                    task_collector_.clone(),
+                    proxy_config.clone(),
+                )
+                .await;
+            }
+            State::Healthy(msg) => {
+                info!("HEALTHY message: {}", msg);
+            }
+        }
+    }
+}
+
+fn kill_tasks(task_collector: Arc<Mutex<Vec<(AbortHandle, String)>>>) {
+    let _ = task_collector.safe_lock(|t| {
+        while let Some(handle) = t.pop() {
+            handle.0.abort();
+            warn!("Killed task: {:?}", handle.1);
+        }
+    });
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn start<'a>(
```

Comment:
ummm ?? could you please why all of these inputs?  how are you expecting other users to call this function?

Reply:
I agree with the criticism, but I wonder if this couldn't be addressed in #1066 , since there's already some refactoring going on there?

Reply:
> ummm ?? could you please why all of these inputs? how are you expecting other users to call this function?

I do not expect other users to call this function.

Reply:
This is a `start` function to the `translator` role, how are u not expecting other users to call it?
I mean, clippy is already screaming and you decide to disable it, I dont think this is the best approach